tools/src/dbgen/Lib.hs

```diff
@@ -172,21 +204,109 @@ generateWalletDB CLI{..} spec@GenSpec{..} = do
 
     -- TODO(ks): Simplify this?
     let fakeUtxoSpec = fakeUtxoCoinDistr walletSpec
+    let fakeTxs      = fakeTxsHistory walletSpec
+
     case addTo of
-        Just accId -> case fakeUtxoSpec of
-            NoDistribution -> addAddressesTo accId spec
-            _              -> generateFakeUtxo fakeUtxoSpec accId
+        Just accId ->
+            if (checkIfAddTo fakeUtxoSpec fakeTxs) then
+                addAddressesTo spec accId
+            else do
+                timed $ generateFakeUtxo fakeUtxoSpec accId
+                timed $ generateFakeTxs fakeTxs accId
+
         Nothing -> do
             say $ printf "Generating %d wallets..." wallets
             wallets' <- timed (forM [1..wallets] genWallet)
             forM_ (zip [1..] wallets') (genAccounts spec)
     say $ green "OK."
+  where
+    checkIfAddTo :: FakeUtxoCoinDistribution -> FakeTxsHistory -> Bool
+    checkIfAddTo NoDistribution NoHistory = True
+    checkIfAddTo _              _         = False
+
+
+-- | Here we generate fake txs. For now it's a simple arbitrary generation.
+generateFakeTxs :: FakeTxsHistory -> AccountId -> UberMonad ()
+generateFakeTxs NoHistory _                = pure ()
+generateFakeTxs SimpleTxsHistory{..} aId   = do
+    -- Get the number of txs we need to generate.
+    let txsNumber = fromIntegral txsCount
+
+    let batchSize = 100
+    let batches   = txsNumber `mod` batchSize
+    let remainder = txsNumber - (batches * batchSize)
+
+    void $ replicateM batches (generateNFakeTxs batchSize aId)
+    generateNFakeTxs remainder aId
+
+-- | Se we can run it in batches so we don't run out of memory. Denis.
+generateNFakeTxs :: Int -> AccountId -> UberMonad ()
+generateNFakeTxs txsNumber aId = do
+
+    db <- askWalletDB
+
+    accounts <- getAccounts Nothing
+
+    let accountsAddrs :: [Address]
+        accountsAddrs = rights . map unwrapCAddress $ concatMap caAddresses accounts
+
+    let outputAddresses :: [Address]
+        outputAddresses = take txsNumber accountsAddrs
+
+    let genTxOut :: Gen [TxOut]
+        genTxOut = forM outputAddresses $ \address -> TxOut address <$> genCoins
+
+    let genTxHistoryEntry :: IO TxHistoryEntry
+        genTxHistoryEntry = generate $ THEntry
```

Comment:
I have two notices for this function:
1. Do I understand correctly that `_thTx` isn't consistent with `_thInputs` and `_thOutputAddrs`? They must be, but it seems to be not important for our current cases.
2. Do I understand correctly that we generate only incoming transactions, not outgoing?

Reply:
1. Yes. I found this working so I thought we didn't need it. Maybe we do?
2. We generate transactions coming in the existing addresses of the accounts of the wallet, yes. Do you think we need to generate outgoing addresses?

Reply:
2. dunno, at least I think you should drop comment about it